### PR TITLE
DATAUP-389 Add application info class

### PR DESCRIPTION
**Pull request by MrCreosote**
# Description of PR purpose/changes

Will be part of a job reqs class in the future

# Jira Ticket / Github Issue #
- [X] Added the Jira Ticket to the title of the PR e.g. (DATAUP-69 Adds a PR template)

# Testing Instructions
* Details for how to test the PR: 
- [X] Tests pass in Github Actions and locally 
- [X] Changes available by spinning up a local test suite

# Dev Checklist:

- [X] My code follows the guidelines at https://sites.google.com/truss.works/kbasetruss/development
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [n/a] I have made corresponding changes to the documentation
- [?] My changes generate no new warnings - 3k warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [n/a] Any dependent changes have been merged and published in downstream modules
- [X] I have run Black and Flake8 on changed Python Code manually or with git precommit (and the Github Actions build passes)

# Updating Version and Release Notes (if applicable)

- [n/a] [Version has been bumped](https://semver.org/) for each release
- [n/a] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: test/tests_for_utils/application_info_test.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+
+def _app_info_init_fail(meth, app, strict, expected):
+    with raises(Exception) as got:
+        AppInfo(meth, app, strict)
+    assert_exception_correct(got.value, expected)
+
+
+def test_equals():
+    assert AppInfo("m.n", "m/p") == AppInfo("m.n", "m/p")
```
Comment (ialarmedalien) [inline test/tests_for_utils/application_info_test.py:173]: how about a test where one of the comparators is a different class?

Comment (MrCreosote) [reply test/tests_for_utils/application_info_test.py:173]: Is this not enough? https://github.com/kbase/execution_engine2/pull/329/files#diff-14dda797937003df3d62117e81a114cd23c0c2eb185a2be7e184b246a33226edR180

Comment (ialarmedalien) [reply test/tests_for_utils/application_info_test.py:173]: I was thinking of creating an instance of a different class, but that also works.

**Review by ialarmedalien — APPROVED** [review]
👍

**Pull request merged**